2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -253,11 +261,17 @@ function mongo_reset_passwords() {
 
 # setup_keyfile fixes the bug in mounting the Kubernetes 'Secret' volume that
 # mounts the secret files with 'too open' permissions.
+# add --keyFile argument to mongo_common_args
 function setup_keyfile() {
+  # If user specify keyFile in cofig file do not use generated keyFile
+  if grep -q "keyFile" ${MONGODB_CONFIG_PATH}; then
```

Comment:
this is a very broad grep... I'd prefer something more specific to avoid matching the string "keyFile" in some arbitrary place of the config, like a value or comment, not a key.